envoy/datadog_checks/envoy/utils.py

```diff
@@ -0,0 +1,20 @@
+from collections import defaultdict
+
+
+def tree():
+    return defaultdict(tree)
+
+
+def make_metric_tree(metrics):
+    metric_tree = tree()
+
+    for metric in metrics:
+        parts = metric.split('.')
+        mapping = metric_tree
```

Comment:
why the copy?

Reply:
We make `mapping` reference the root of the tree at every iteration to create the new branches.